src/utils/validations/component/validations/form-with-validations.hoc.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+const ValidationsContext = React.createContext();
+
+const withValidations = (WrappedComponent) => {
+  return class extends React.Component {
+    state = {
+      errorCount: 0,
+      warningCount: 0,
+      infoCount: 0
+    }
+
+    inputs = {};
+
+    validationTypes = ['error', 'warning', 'info'];
+
+    addInput = (name, validate) => {
+      this.inputs[name] = validate;
+    }
+
+    removeInput = (name) => {
+      delete this.inputs[name];
+    }
+
+    adjustCount = (type, validationResult) => {
+      if (this.validationTypes.includes(type)) {
+        const stateProp = `${type}Count`;
+        let adjustment = -1;
+
+        if (validationResult) adjustment = 1;
+        else if (this.state[stateProp] === 0) adjustment = 0;
+
+        this.setState((prev) => { return { [stateProp]: prev[stateProp] + adjustment }; });
+      }
+    }
+
+    getContext = () => {
+      return {
+        addInput: this.addInput,
+        removeInput: this.removeInput,
+        adjustCount: this.adjustCount
+      };
+    }
+
+    validateRegisteredInputs = () => {
+      let results;
+      return new Promise((resolve, reject) => {
+        results = Object.keys(this.inputs).map((name) => {
```

Comment:
`map` isn't really the correct method to do here.

Consider using `for..in` instead.